## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#743d8fe3-c2b9-4511-afb4-ca5629d28f14) |
| mint-xl.sh | :x: [more...](#fb2d79d5-4b3e-4a43-92c7-289baba8c608) |
| mint-large-bucket.sh | :x: [more...](#98b44da8-6471-4277-872c-1d9ade6d57d2) |
| mint-compression-fs.sh | :x: [more...](#db875b5d-af1f-4ea8-8e10-cd190acfb580) |
<a name="fb2d79d5-4b3e-4a43-92c7-289baba8c608"></a>
### 8459-2e008c0/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:31134
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ba255df82a39:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-1zr54ybiqcmfi1",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366189,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D1F2F06979B794, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="98b44da8-6471-4277-872c-1d9ade6d57d2"></a>
### 8459-2e008c0/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:31003
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 37fd7594e6de:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 5 seconds
(2/14) Running aws-sdk-java tests ... done in 3 seconds
(3/14) Running aws-sdk-php tests ... done in 51 seconds
(4/14) Running aws-sdk-ruby tests ... done in 12 seconds
(5/14) Running awscli tests ... done in 1 minutes and 29 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 30 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 34 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 50 seconds
{
  "args": {},
  "duration": 106494,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="743d8fe3-c2b9-4511-afb4-ca5629d28f14"></a>
### 8459-2e008c0/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30912
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 76ca77829a36:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-sara6pbs1vxgz9",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364229,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D1F2ED11E5D7FF, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="db875b5d-af1f-4ea8-8e10-cd190acfb580"></a>
### 8459-2e008c0/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:32549
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0ba48cda30a9:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 4 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 55 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 31 seconds
(8/14) Running minio-dotnet tests ... done in 34 seconds
(9/14) Running minio-go tests ... done in 55 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 48 seconds
(11/14) Running minio-js tests ... done in 44 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 13 seconds
{
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "duration": 140,
  "status": "FAIL",
  "args": {
    "object_name": "0b0305a2-2f22-45b5-bbda-68a25286dd3c",
    "presigned_get_object_url": "http://72.28.97.51:32549/minio-py-test-dc80bc90-e8e9-4529-85ce-f45e3b0b3e44/0b0305a2-2f22-45b5-bbda-68a25286dd3c?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191028%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191028T232509Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=e1fd389ce89afa94a4201eb936e9790a5bb57a840afba9a183485498f9789dba",
    "response.reason": "Forbidden",
    "response.status": 403,
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>0b0305a2-2f22-45b5-bbda-68a25286dd3c</Key><BucketName>minio-py-test-dc80bc90-e8e9-4529-85ce-f45e3b0b3e44</BucketName><Resource>/minio-py-test-dc80bc90-e8e9-4529-85ce-f45e3b0b3e44/0b0305a2-2f22-45b5-bbda-68a25286dd3c</Resource><RequestId>15D1F32D97A8D2CC</RequestId><HostId>6c162cab-17ad-4e06-b428-5ec9100f63be</HostId></Error>",
    "bucket_name": "minio-py-test-dc80bc90-e8e9-4529-85ce-f45e3b0b3e44",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-10-28T22-55-34Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15D1F32D97A8D2CC\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Mon, 28 Oct 2019 23:25:22 GMT\"]}}"
  },
  "message": "AccessDenied: message: Access Denied",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n"
}

Executed 11 out of 14 tests successfully.
```
Error: No such image: minio/minio:8459-2e008c0

--- a/cmd/bucket-handlers-listobjects.go
+++ b/cmd/bucket-handlers-listobjects.go
@@ -128,7 +128,7 @@ func (api objectAPIHandlers) ListBucketObjectVersionsHandler(w http.ResponseWrit
 
 // ListObjectsV2Handler - GET Bucket (List Objects) Version 2.
 // --------------------------
-// This implementation of the GET operation returns some or all (up to 1000)
+// This implementation of the GET operation returns some or all (up to 10000)
 // of the objects in a bucket. You can use the request parameters as selection
 // criteria to return a subset of the objects in a bucket.
 //
@@ -210,7 +210,7 @@ func (api objectAPIHandlers) ListObjectsV2Handler(w http.ResponseWriter, r *http
 
 // ListObjectsV1Handler - GET Bucket (List Objects) Version 1.
 // --------------------------
-// This implementation of the GET operation returns some or all (up to 1000)
+// This implementation of the GET operation returns some or all (up to 10000)
 // of the objects in a bucket. You can use the request parameters as selection
 // criteria to return a subset of the objects in a bucket.
 //

--- a/cmd/config.go
+++ b/cmd/config.go
@@ -51,7 +51,7 @@ func listServerConfigHistory(ctx context.Context, objAPI ObjectLayer) ([]madmin.
 	// List all kvs
 	marker := ""
 	for {
-		res, err := objAPI.ListObjects(ctx, minioMetaBucket, minioConfigHistoryPrefix, marker, "", 1000)
+		res, err := objAPI.ListObjects(ctx, minioMetaBucket, minioConfigHistoryPrefix, marker, "", maxObjectList)
 		if err != nil {
 			return nil, err
 		}

--- a/cmd/daily-lifecycle-ops.go
+++ b/cmd/daily-lifecycle-ops.go
@@ -135,7 +135,7 @@ func lifecycleRound(ctx context.Context, objAPI ObjectLayer) error {
 		// List all objects and calculate lifecycle action based on object name & object modtime
 		marker := ""
 		for {
-			res, err := objAPI.ListObjects(ctx, bucket.Name, commonPrefix, marker, "", 1000)
+			res, err := objAPI.ListObjects(ctx, bucket.Name, commonPrefix, marker, "", maxObjectList)
 			if err != nil {
 				continue
 			}

--- a/cmd/iam-object-store.go
+++ b/cmd/iam-object-store.go
@@ -540,7 +540,7 @@ func listIAMConfigItems(objectAPI ObjectLayer, pathPrefix string, dirs bool,
 		marker := ""
 		for {
 			lo, err := objectAPI.ListObjects(context.Background(),
-				minioMetaBucket, pathPrefix, marker, SlashSeparator, 1000)
+				minioMetaBucket, pathPrefix, marker, SlashSeparator, maxObjectList)
 			if err != nil {
 				select {
 				case ch <- itemOrErr{Err: err}:

--- a/cmd/web-handlers.go
+++ b/cmd/web-handlers.go
@@ -418,7 +418,8 @@ func (web *webAPIHandlers) ListObjects(r *http.Request, args *ListObjectsArgs, r
 		nextMarker := ""
 		// Fetch all the objects
 		for {
-			result, err := core.ListObjects(args.BucketName, args.Prefix, nextMarker, SlashSeparator, 1000)
+			result, err := core.ListObjects(args.BucketName, args.Prefix, nextMarker, SlashSeparator,
+				maxObjectList)
 			if err != nil {
 				return toJSONError(ctx, err, args.BucketName)
 			}
@@ -532,7 +533,7 @@ func (web *webAPIHandlers) ListObjects(r *http.Request, args *ListObjectsArgs, r
 	nextMarker := ""
 	// Fetch all the objects
 	for {
-		lo, err := listObjects(ctx, args.BucketName, args.Prefix, nextMarker, SlashSeparator, 1000)
+		lo, err := listObjects(ctx, args.BucketName, args.Prefix, nextMarker, SlashSeparator, maxObjectList)
 		if err != nil {
 			return &json2.Error{Message: err.Error()}
 		}
@@ -711,7 +712,7 @@ next:
 		marker := ""
 		for {
 			var lo ListObjectsInfo
-			lo, err = listObjects(ctx, args.BucketName, objectName, marker, "", 1000)
+			lo, err = listObjects(ctx, args.BucketName, objectName, marker, "", maxObjectList)
 			if err != nil {
 				break next
 			}
@@ -1363,7 +1364,8 @@ func (web *webAPIHandlers) DownloadZip(w http.ResponseWriter, r *http.Request) {
 		// date to the response writer.
 		marker := ""
 		for {
-			lo, err := listObjects(ctx, args.BucketName, pathJoin(args.Prefix, object), marker, "", 1000)
+			lo, err := listObjects(ctx, args.BucketName, pathJoin(args.Prefix, object), marker, "",
+				maxObjectList)
 			if err != nil {
 				return
 			}

--- a/cmd/xl-sets.go
+++ b/cmd/xl-sets.go
@@ -1620,7 +1620,7 @@ func (s *xlSets) HealObjects(ctx context.Context, bucket, prefix string, healObj
 			}
 		}
 
-		res, err := s.ListObjectsHeal(ctx, bucket, prefix, marker, "", 1000)
+		res, err := s.ListObjectsHeal(ctx, bucket, prefix, marker, "", maxObjectList)
 		if err != nil {
 			continue
 		}

